aether-ui/aether/ui/api/utils.py

```diff
@@ -661,3 +662,23 @@ def kernel_data_request(url='', method='get', data=None, headers=None):
 
 def wrap_kernel_headers(instance):
     return mt_utils.add_instance_realm_in_headers(instance, kernel_utils.get_kernel_auth_header())
+
+
+def delete_operation(url, data, object):
+    try:
+        response = kernel_data_request(
+            url=url,
+            method='post',
+            data=data,
+            headers=wrap_kernel_headers(object),
+        )
+        object.delete()
+        return Response(response)
```

Comment:
I don't like so much the idea of returning a `Response` object in a utility method, because this cannot be reused later in another context... why don't let the view method handle the response statuses...

```python
    try:
        response = kernel_data_request(
            url=url,
            method='post',
            data=data,
            headers=wrap_kernel_headers(obj),
        )
        obj.delete()
        return response

    except HTTPError as e:
        if e.response.status_code == status.HTTP_404_NOT_FOUND:
            obj.delete()
            return None
        raise e
```

In the view:

```python
    @action(detail=True, methods=['post'], url_path='delete-artefacts')
    def delete_artefacts(self, request, pk=None, *args, **kwargs):
        pipeline = self.get_object_or_404(pk=pk)
        
        artefacts_result = {'not_published': True}  # default response
        if pipeline.mappingset:
            try:
                artefacts_result = utils.delete_operation(
                    f'mappingsets/{pipeline.mappingset}/delete-artefacts/',
                    request.data,
                    pipeline,
                )
            except HTTPError as http_e:
                return Response(str(http_e), status=http_e.response.status_code)
            except Exception as e:
                return Response(str(e), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
        else:
            pipeline.delete()
        
        return Response(artefacts_result)
```